tasks/setup_environment.yml

```diff
@@ -0,0 +1,90 @@
+---
+- name: Set project and service identifier in facts
+  ansible.builtin.set_fact:
+    project: "{{ group_names[0] }}"
+    service_identifier: >-
+      {{ group_names[0] }}{{ '-' + client if type in default_node_types and client is defined else '-' + type + ('-' + client if client is defined else '') }}
+
+- name: Set architecture variable based on system architecture
+  ansible.builtin.set_fact:
+    system_architecture: "{{ architecture_map[ansible_architecture] | default(ansible_architecture) }}"
+
+- name: Display environment being deployed to
+  ansible.builtin.debug:
+    msg:
+      - "Deploying to Host: {{ inventory_hostname }}"
+      - "Groups: {{ group_names }}"
+      - "Project: {{ project }}"
+      - "Environment: {{ env }}"
+      - "Type: {{ type }}"
+      - "Version: {{ vars[project][type][client].version }}"
+      - "Username: {{ service_identifier }}"
+      - "Service Name: {{ service_identifier }}"
+      - "Operating System: {{ ansible_system | lower }}"
+      - "System Architecture: {{ system_architecture }}"
+
+- name: Set all port variables dynamically for consensus client
+  ansible.builtin.set_fact:
+    p2p_port: "{{ vars[project][type].port_prefix ~ vars[project][type][client].ports.suffix.p2p }}"
+    rpc_port: "{{ vars[project][type].port_prefix ~ vars[project][type][client].ports.suffix.rpc }}"
+    abci_port: "{{ vars[project][type].port_prefix ~ vars[project][type][client].ports.suffix.abci }}"
+    prometheus_port: "{{ vars[project][type].port_prefix ~ vars[project][type][client].ports.suffix.prometheus }}"
+    pprof_port: "{{ vars[project][type].port_prefix ~ vars[project][type][client].ports.suffix.pprof }}"
+    rest_api_port: "{{ vars[project][type].port_prefix ~ vars[project][type][client].ports.suffix.rest_api }}"
+    rosetta_port: "{{ vars[project][type].port_prefix ~ vars[project][type][client].ports.suffix.rosetta }}"
+    grpc_port: "{{ vars[project][type].port_prefix ~ vars[project][type][client].ports.suffix.grpc }}"
+    grpc_web_port: "{{ vars[project][type].port_prefix ~ vars[project][type][client].ports.suffix.grpc_web }}"
+    execution_engine_port: "{{ vars[project][type].port_prefix ~ vars[project][type].execution.ports.suffix.engine }}"
```

Comment:
Its added under execution ports already, any reason to add it here again ?

Reply:
consensus client needs to connect to execution client.

Reply:
yes, but its being redefined here as well https://github.com/encapsulate-xyz/berachain-ansible/pull/1/files/e30a74ec5328970fff400cc32539f2186b89b904#diff-5b326a46b98bf64dd1ef268b3e6f8633c26391fa8f091723afbe46ff84d32f4dR42 

its okay lets keep it in both places I guess